frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.unit.spec.tsx

```diff
@@ -154,4 +129,46 @@ describe("FilterPopover", () => {
       });
     });
   });
+  describe("filter rendering", () => {
+    describe("no-value filters: %s filter", () => {
+      it.each(["is-null", "not-null", "is-empty", "not-empty"])(
+        "should not render picker or separator when selecting a new filter from the column dropdown",
+        async operator => {
+          setup({
+            filter: [operator, ["field", PRODUCTS.TITLE, null], null] as Filter,
+            showFieldPicker: false,
+          });
+
+          expect(
+            screen.queryByTestId("filter-popover-separator"),
+          ).not.toBeInTheDocument();
```

Comment:
It's hard for me to feel confident about tests without asserting certain UI exists. For example, if this component is rendering asynchronously e.g. relying on an API call and displaying the loading spinner, the test would have passed before those UIs that aren't supposed to be rendered are rendered in later render cycle.

Could you add some assertions on the existing UI before this check? It should look something like
```
assertSomethingShouldBeVisible
assertSomethingShouldNotBeRendered
assertSomethingShouldNotBeRendered
```

Reply:
I ended up switching the tests to check that the "empty" `div` is visible, rather than checking the absence of those two elements